lunavl/sdk/detectors/humandetector.py

```diff
@@ -0,0 +1,237 @@
+        return toDetect
+
+    @CoreExceptionWrap(LunaVLError.DetectHumanError)
+    def detectOne(
+        self, image: VLImage, detectArea: Optional[Rect] = None, detectLandmarks: bool = True
+    ) -> Union[None, HumanDetection]:
+        """
+        Detect just one best detection on the image.
+
+        Args:
+            image: image. Format must be R8G8B8
+            detectArea: rectangle area which contains human to detect. If not set will be set image.rect
+            detectLandmarks: detect or not landmarks
+        Returns:
+            human detection if human is found otherwise None
+        Raises:
+            LunaSDKException: if detectOne is failed or image format has wrong  the format
+        """
+        assertImageForDetection(image)
+
+        if detectArea is None:
+            forDetection = ImageForDetection(image=image, detectArea=image.rect)
+        else:
+            forDetection = ImageForDetection(image=image, detectArea=detectArea)
+        detections = self.detect([forDetection], limit=1, detectLandmarks=detectLandmarks)
+        if detections[0]:
+            return detections[0][0]
+        return None
+
+    @CoreExceptionWrap(LunaVLError.DetectHumansError)
+    def detect(
+        self, images: List[Union[VLImage, ImageForDetection]], limit: int = 5, detectLandmarks: bool = True
+    ) -> List[List[HumanDetection]]:
+        """
+        Batch detect human bodies on images.
+
+        Args:
+            images: input images list. Format must be R8G8B8
+            limit: max number of detections per input image
+            detectLandmarks: detect or not landmarks
+        Returns:
+            return list of lists detection, order of detection lists is corresponding to order input images
+        Raises:
+            LunaSDKException(LunaVLError.InvalidImageFormat): if any image has bad format or detect is failed
+
+        """
+        imgs, detectAreas = getArgsForCoreDetectorForImages(images)
+
+        error, detectRes = self._detector.detect(imgs, detectAreas, limit, self._getDetectionType(detectLandmarks))
+        assertError(error)
+        res = []
+        for numberImage, imageDetections in enumerate(detectRes):
+            image_ = images[numberImage]
+            image = image_ if isinstance(image_, VLImage) else image_.image
+            res.append([HumanDetection(coreDetection, image) for coreDetection in imageDetections])
+        return res
+
+    @overload
+    def redetectOne(self, image: VLImage, *, bBox: Optional[Rect]) -> Union[None, HumanDetection]:  # noqa: F811
+        ...
+
+    @overload
+    def redetectOne(  # noqa: F811
+        self, image: VLImage, *, detection: Optional[HumanDetection]
+    ) -> Union[None, HumanDetection]:
+        ...
+
+    @CoreExceptionWrap(LunaVLError.DetectHumansError)
+    def redetectOne(  # noqa: F811
+        self, image: VLImage, *, bBox: Optional[Rect] = None, detection: Optional[HumanDetection] = None
```

Comment:
why not to use isinstance? you need to cover only two argumetns. It seems more difficult with @overload and three function definitions...

Reply:
i want that interface of human detector will be similar on face detector. Change interface for face and human detection